fdbclient/BackupContainerFileSystem.actor.cpp

```diff
@@ -2395,4 +2395,109 @@ TEST_CASE("/backup/containers/localdir/missingLogRangesRestorability") {
+		if (e.code() != error_code_backup_invalid_url && e.code() != error_code_backup_does_not_exist)
+			throw;
+	}
+
+	wait(c->create());
+
+	state int blockSize = 1024;
+	state std::vector<std::string> rangeFileNames;
+	state Key begin = randomKeyBetween(normalKeys);
+	state Key end = randomKeyBetween(KeyRangeRef(begin, normalKeys.end));
+	state std::pair<Key, Key> beginEndKeys = std::make_pair(begin, end);
+	state std::vector<std::pair<Key, Key>> snapshotBeginEndKeys;
+
+	// writing random number of range files with rangeSize 100
+	state std::vector<Future<Void>> writes;
+	state Version snapshotBeginVersion = 10;
+	state Version snapshotEndVersion = deterministicRandom()->randomInt(500, 1000);
+	state Version rangeSize = 100;
+	state Version v = snapshotBeginVersion;
+	state int numRangeFiles = 0;
+	while (v <= snapshotEndVersion) {
+		state Reference<IBackupFile> range = wait(c->writeRangeFile(v, 0, v, blockSize));
+		writes.push_back(writeAndVerifyFile(c, range, 100, &lock));
+		rangeFileNames.push_back(range->getFileName());
+		snapshotBeginEndKeys.push_back(beginEndKeys);
+		v += rangeSize;
+		++numRangeFiles;
+	}
+	snapshotEndVersion = v - rangeSize;
+
+	// writing random number of log files with logSize 70, covering the entire snapshot
+	state Version logSize = 70;
+	v = snapshotBeginVersion;
+	state int numLogFiles = 0;
+	while (v <= snapshotEndVersion) {
+		Reference<IBackupFile> log = wait(c->writeLogFile(v, v + logSize, blockSize));
+		writes.push_back(writeAndVerifyFile(c, log, 100, &lock));
+		++numLogFiles;
+		v += logSize;
+	}
+
+	// writing snapshot file
+	writes.push_back(c->writeKeyspaceSnapshotFile(
+	    rangeFileNames, snapshotBeginEndKeys, deterministicRandom()->randomInt(0, 2e6), IncludeKeyRangeMap(BUGGIFY)));
+	wait(waitForAll(writes));
+
+	state BackupFileList fileList = wait(c->dumpFileList());
+	printFileList(fileList);
+	ASSERT_EQ(fileList.ranges.size(), numRangeFiles);
+	ASSERT_EQ(fileList.logs.size(), numLogFiles);
+	ASSERT_EQ(fileList.snapshots.size(), 1);
+
+	state BackupDescription desc = wait(c->describeBackup());
+	printf("\n%s\n", desc.toString().c_str());
+	ASSERT_EQ(desc.minLogBegin, snapshotBeginVersion);
+	ASSERT_EQ(desc.maxLogEnd, v);
+	ASSERT_EQ(desc.minRestorableVersion, snapshotEndVersion);
+	ASSERT_EQ(desc.maxRestorableVersion, v - 1);
+	ASSERT_EQ(desc.snapshots[0].restorable, true);
+	ASSERT_EQ(desc.contiguousLogEnd, v);
+
+	// writing random number of more continuous log files
+	state int newNumLogFiles = deterministicRandom()->randomInt(2, 8);
+	writes.clear();
+	while (newNumLogFiles) {
+		Reference<IBackupFile> log = wait(c->writeLogFile(v, v + logSize, blockSize));
+		writes.push_back(writeAndVerifyFile(c, log, 100, &lock));
+		--newNumLogFiles;
+		v += logSize;
+	}
+	wait(waitForAll(writes));
+
+	state BackupFileList fileList1 = wait(c->dumpFileList());
+	printFileList(fileList1);
+	ASSERT_EQ(fileList.ranges.size(), numRangeFiles);
+	ASSERT_EQ(fileList.logs.size(), numLogFiles + newNumLogFiles);
```

Comment:
`newNumLogFiles` is 0 now. I suppose `numLogFiles` should increment by `newNumLogFiles` right after its initialization.

Reply:
My bad, the test did not complain because I was using fileList instead of fileList1 for below asserts. 
Updated the PR.